src/main/java/org/example/Game.java

```diff
@@ -1,181 +1,262 @@
+      isWinningMoveOrTie(player.getSymbol());
+    }
+  }
 
-            } catch (InputMismatchException e) {
-                System.out.println("Plzz Enter only numbers  from 1 to 9 !! ");
+  private void printUserMove(String name) {
+    String result =
+        String.format("~~~~~~~~~~~~~~~~~~~~~~~~~~~  %s's Move  ~~~~~~~~~~~~~~~~~~~~~~~~~~", name);
+    println(result);
+    boardobj.showBoard();
+  }
 
-            } finally {
-                sc.nextLine();
-            }
+  private void removePositionFromPositionList(int position) {
+    boardobj.getEmptyPositions().remove(Integer.valueOf(position));
+  }
 
-        } while (!("Integer".equals(((Object) position).getClass().getSimpleName())) || !(position >= 1 && position <= 9) || !boardobj.getIntList().contains(position));
+  private void removePositionFromPositionList() {
 
+    boardobj.getEmptyPositions().remove(0);
+  }
 
-        int row = (position - 1) / 3;
-        int col = position - (row * 3) - 1;
+  private void manual() {
+    println(
+        "\uD83C\uDF69 \uD83C\uDF69 Welcome to the classical Tic-Tac-Toe game \uD83C\uDF69 \uD83C\uDF69");
+    println(
+        "\uD83D\uDCA1 Start the game by placing Xs and Os alternatively\n"
+            + "\uD83D\uDCA1 The goal is to get three of your symbols in a row, column or diagonal \n"
+            + "\uD83D\uDCA1 The game ends when one player wins or the grid is full, resulting in a TIE\n"
+            + "Happy Gaming \uD83C\uDFAE");
+  }
 
-        boardobj.getBoard()[row][col] = player.getSymbol();
+  private void rePlay() {
+    postStartup();
+  }
 
-        int pos = boardobj.getIntList().indexOf(position);
-        boardobj.getIntList().remove(pos);
-        String result = String.format("~~~~~~~~~~~~~~~~~~~~~~~~~~~  %s's Move  ~~~~~~~~~~~~~~~~~~~~~~~~~~", player.getName());
-        System.out.println(result);
-        boardobj.showBoard();
-        isWinningMoveOrTie(player.getSymbol());
+  private void menu() {
+    println(
+        "\uD83D\uDCA1 If u wanna replay ♻\uFE0F then press R else press aything ( except power button \uD83D\uDE05 )and Enter to quit ⛩\uFE0F !! ");
 
+    String pref = this.sc.nextLine().toUpperCase();
+    if (!pref.isEmpty() && 'R' == pref.charAt(0)) {
+      rePlay();
+    } else {
+      exit(0);
     }
+  }
 
-    private void manual() {
-        System.out.println("\uD83C\uDF69 \uD83C\uDF69 Welcome to the classical Tic-Tac-Toe game \uD83C\uDF69 \uD83C\uDF69");
-        System.out.println("\uD83D\uDCA1 Start the game by placing Xs and Os alternatively\n" +
-                "\uD83D\uDCA1 The goal is to get three of your symbols in a row, column or diagonal \n" +
-                "\uD83D\uDCA1 The game ends when one player wins or the grid is full, resulting in a TIE\n" +
-                "Happy Gaming \uD83C\uDFAE");
+  private void checkRowCol(char[][] board, char symbol) {
+    for (int i = 0; i < 3; i++) { // checking row and column trio
```

Comment:
Try not to use 3 as a constant variable. Make it extensible for nXn by writing their BOARD_SIZE.

Reply:
done